clang-tools-extra/clang-doc/Serialize.cpp

```diff
@@ -739,6 +757,24 @@ static void populateSymbolInfo(SymbolInfo &I, const T *D, const FullComment *C,
     I.Loc.emplace_back(Loc);
 }
 
+static void populateConstraints(TemplateInfo &I, const TemplateDecl *D) {
+  if (!D || !D->hasAssociatedConstraints())
+    return;
+
+  SmallVector<AssociatedConstraint> AssociatedConstraints;
+  D->getAssociatedConstraints(AssociatedConstraints);
+  if (AssociatedConstraints.size() == 1) {
+    auto *ConstraintExpr = dyn_cast<ConceptSpecializationExpr>(
+        AssociatedConstraints.front().ConstraintExpr);
+    ConstraintInfo CI(getUSRForDecl(ConstraintExpr->getNamedConcept()),
+                      ConstraintExpr->getNamedConcept()->getNameAsString());
+    CI.Expression = exprToString(ConstraintExpr);
+    I.Constraints.push_back(std::move(CI));
+    return;
+  }
+  // FIXME: Handle compound constraints.
```

Comment:
Would love some opinions on how to handle multiple constraints. Since they can be conjunctive or disjunctive, their ordering is pretty important. I'd assume we want to preserve their order/relationships somehow but not sure how high that is on the prio list. My initial idea would be having a "super" object that denotes a conjunctive or disjunctive relationship.

Reply:
That could be OK.  Just tracking them in program order for now, seems like a good place to start. I'd rather not have to keep a syntax tree around for these, but we may have to. I imagine if we did that the. "syntax tree" would be a small/simple approximation for the boolean logic and statements. That all sounds gross just typing it out though ...